internal/self_metrics/self_metrics.go

```diff
@@ -99,7 +99,7 @@ func InstrumentEnabledReceiversMetric(ctx context.Context, uc *confgenerator.Uni
 }
 
 func InstrumentFeatureTrackingMetric(uc *confgenerator.UnifiedConfig, meter metricapi.Meter) error {
-	features, err := confgenerator.ExtractFeatures(uc)
+	features, err := confgenerator.ExtractFeatures(context.Background(), uc)
```

Comment:
I think you should pass the context from `CollectOpsAgentSelfMetrics` as an argument.